Add tutorial + example SDF for shadow texture size (#2597)
gazebosim/gz-sim: tutorials/shadow_texture_size.md

\page shadow_texture_size Shadow Texture Size

This tutorial showcases how to change the texture size of shadows casted by lights.

<div style="text-align:center;">
  \image html files/shadow_texture_size/shadow_texsize_2k_to_16k.gif width=60%
</div>

## Texture size options

> Supported texture sizes include: 512 px, 1024 px, 2048 px, 4096 px, 8192 px, 16384 px.

Reviewer: ```suggestion
Supported texture sizes include: 512px, 1024px, 2048px, 4096px, 8192px, 16384px.
```

The default shadow texture size for all lights is 2K. The default max texture size is 16K, but for less powerful computers, it may be a lower value, like 8K.

## Changing shadow texture size

The shadow texture size can be changed for the GUI with the usage of the Minimal Scene GUI plugin in the SDF.

### Example usage for the GUI

> Let's change the shadow texture size for directional light with the SDF file below. (The finished SDF file can be viewed [here](https://github.com/gazebosim/gz-sim/blob/main/examples/worlds/shadow_texture_size.sdf).)

Reviewer: can you mention that we currently only support changing shadow texture size for directional lights?

Author: Yup = done here b48725b50d1b4fd554ec15e56d9c8e9d8d6c6823

Reviewer: oh can you also change the link to point to gz-sim9 branch? i.e. https://github.com/gazebosim/gz-sim/tree/gz-sim9/examples/worlds/shadow_texture_size.sdf

> 1) Save the below in an SDF file named `shadow_texsize_demo.sdf`:

Reviewer: Change the name to be consistent with finished SDF file in `examples/worlds`?

```suggestion
1) Save the below in an SDF file named `shadow_texture_size.sdf`:
```

      <direction>-2 2 -1.5</direction>
      <intensity>1.0</intensity>
    </light>

    <!--scene objects-->
    <include>
      <pose>0 0 0 0 0 -1.57</pose>
      <uri>
      https://fuel.gazebosim.org/1.0/OpenRobotics/models/Garden Mascot
      </uri>
    </include>

    <model name="floor">
    <pose>-5 0 -0.5 0 0 0</pose>
    <static>true</static>
    <link name="link">
      <collision name="collision">
      <geometry>
        <box>
        <size>15 15 1</size>
        </box>
      </geometry>
      </collision>
      <visual name="visual">
      <geometry>
        <box>
        <size>15 15 1</size>
        </box>
      </geometry>
      <material>
        <ambient>0.8 0.8 0.8 1</ambient>
        <diffuse>0.8 0.8 0.8 1</diffuse>
        <specular>0.8 0.8 0.8 1</specular>
      </material>
      </visual>
    </link>
    </model>
  </world>
</sdf>
```

2) Add the following lines to the Minimal Scene GUI plugin.

```xml
</shadows>
  <texture_size light_type="directional">8192</texture_size>
</shadows>
```

The `<texture_size>` value can be changed as per the Texture size options section.

3) Open the `shadow_texsize_demo.sdf` world with

```bash

> gz sim shadow_texsize_demo.sdf

Reviewer: ```suggestion
gz sim shadow_texture_size.sdf
```
